backend/builder/tasks/tasks_build_image.py

```diff
@@ -14,21 +14,18 @@
 
 @app.task(
     bind=True,
-    acks_late=True,
-    reject_on_worker_lost=True,
-    ignore_result=False,
     base=BuildTask,
 )
 # Ack late and reject on worker lost allows use to
 # see http://docs.celeryproject.org/en/latest/userguide/configuration.html#task-reject-on-worker-lost
 # and https://github.com/celery/celery/issues/5106
-def build_image(task: BuildTask, function_serialized: str, channel: str) -> str:
+def build_image(task: BuildTask, function_serialized: str, channel_name: str) -> None:
     function = orchestrator.Function.parse_raw(function_serialized)
 
     attempt = 0
     while attempt <= task.max_retries:
         try:
-            build_image_if_missing(channel, function)
+            build_image_if_missing(channel_name, function)
```

Comment:
Not to be changed here, but this logic is pretty convoluted, could you add a `# TODO refactor` comment so we have a look into it later. 
I feel like we might be taking the problem from the wrong direction

Reply:
All in for adding a todo, this should be tackled in the function build ordering. This is a workaround, as pushing new tasks put it at the end of the queue, we do the retry inside the task